test/index.spec.js

```diff
@@ -319,6 +319,38 @@ describe('daemon client', function () {
         await expect(client.connect(identify.peerId, identify.addrs)).to.be.rejectedWith(
           'unspecified')
       })
+
+      it('should error if it receives an undefined response', async () => {
+        client = new Client(defaultMultiaddr)
+        await client.attach()
+        const identify = await client.identify()
+        client.close()
+
+        client = new Client(addr2)
+        await client.attach()
+
+        stub = sinon.stub(client, 'send').returns({
+          first: () => { return undefined }
+        })
+        await expect(client.connect(identify.peerId, identify.addrs)).to.be.rejectedWith(
+          'unspecified')
+      })
+
+      it('should error if it receives an empty response', async () => {
+        client = new Client(defaultMultiaddr)
+        await client.attach()
+        const identify = await client.identify()
+        client.close()
+
+        client = new Client(addr2)
+        await client.attach()
+
+        stub = sinon.stub(client, 'send').returns({
+          first: () => { return '' }
```

Comment:
A message being `''`, will go into the initial validation: `if (!message) `.

I believe you want to send an errResponse message with an undefined `msg` instead?

Reply:
@vasco-santos This is actually what I want, to test for the case where the response is `''`. It should hit the validation on line 164, so `Response.decode` isn't called. I think `Response.decode` might actually fail if the input is empty.

Reply:
You are right, the other test already exists! 😅